src/pages/workspaces/workspace/applications/AppLauncher.js

```diff
@@ -21,14 +21,39 @@ const ApplicationLauncher = _.flow(
     breadcrumbs: props => breadcrumbs.commonPaths.workspaceDashboard(props),
     title: _.get('application')
   })
-)(({ namespace, name, refreshRuntimes, runtimes, persistentDisks, application, workspace }, ref) => {
+)(({ namespace, name, refreshRuntimes, runtimes, persistentDisks, application, workspace, workspace: { workspace: { googleProject, bucketName } } }, ref) => {
   const cookieReady = Utils.useStore(cookieReadyStore)
   const [showCreate, setShowCreate] = useState(false)
   const [busy, setBusy] = useState(false)
+  const [isWelderInit, setIsWelderInit] = useState(application !== 'RStudio') //we've already init welder if app isn't rstudio
 
   const runtime = getCurrentRuntime(runtimes)
   const runtimeStatus = getConvertedRuntimeStatus(runtime) // preserve null vs undefined
 
+  useEffect(() => {
+    const runtime = getCurrentRuntime(runtimes)
+    const runtimeStatus = getConvertedRuntimeStatus(runtime)
+
+    const setupWelder = _.flow(
+      Utils.withBusyState(setBusy),
+      withErrorReporting('Error setting up analysis')
+    )(async () => {
+      const localBaseDirectory = ``
+      const localSafeModeBaseDirectory = ``
```

Comment:
`localSafeModeBaseDirectory` is now optional in this API request once this [welder PR](https://github.com/DataBiosphere/welder/pull/68) is merged and integrated with Leo. We don't need `localSafeModeBaseDirectory` for Rmd file syncing so I would remove that from the Ajax call

Reply:
I'd prefer to include explicitly since we have tickets to possibly use it down the line, to simplify the API from the UI's perspective (since Jupyter still needs it)

Reply:
```suggestion
      const localSafeModeBaseDirectory = ''
```